recipes/assimp/5.0.x/patches/0001-unvendor-deps.patch

```diff
@@ -0,0 +1,176 @@
+--- a/code/3MF/D3MFExporter.cpp
++++ b/code/3MF/D3MFExporter.cpp
+@@ -58,7 +58,7 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ #ifdef ASSIMP_USE_HUNTER
+ #  include <zip/zip.h>
+ #else
+-#  include <contrib/zip/src/zip.h>
++#  include <zip/zip.h>
+ #endif
+ 
+ namespace Assimp {
+--- a/code/Blender/BlenderTessellator.h
++++ b/code/Blender/BlenderTessellator.h
+@@ -147,7 +147,7 @@ namespace Assimp
+ #ifdef ASSIMP_USE_HUNTER
+ #  include <poly2tri/poly2tri.h>
+ #else
+-#  include "../contrib/poly2tri/poly2tri/poly2tri.h"
++#  include <poly2tri/poly2tri.h>
+ #endif
+ 
+ namespace Assimp
+--- a/code/CMakeLists.txt
++++ b/code/CMakeLists.txt
+@@ -893,7 +893,7 @@ ENDIF(HUNTER_ENABLED)
+ IF(HUNTER_ENABLED)
+   hunter_add_package(poly2tri)
+   find_package(poly2tri CONFIG REQUIRED)
+-ELSE(HUNTER_ENABLED)
++ELSEIF(0)
+   SET( Poly2Tri_SRCS
+     ../contrib/poly2tri/poly2tri/common/shapes.cc
+     ../contrib/poly2tri/poly2tri/common/shapes.h
+@@ -914,7 +914,7 @@ ENDIF(HUNTER_ENABLED)
+ IF(HUNTER_ENABLED)
+   hunter_add_package(minizip)
+   find_package(minizip CONFIG REQUIRED)
+-ELSE(HUNTER_ENABLED)
++ELSEIF(0)
+   SET( unzip_SRCS
+     ../contrib/unzip/crypt.h
+     ../contrib/unzip/ioapi.c
+@@ -929,7 +929,7 @@ ENDIF(HUNTER_ENABLED)
+ IF(HUNTER_ENABLED)
+   hunter_add_package(zip)
+   find_package(zip CONFIG REQUIRED)
+-ELSE(HUNTER_ENABLED)
++ELSEIF(0)
+   SET( ziplib_SRCS
+     ../contrib/zip/src/miniz.h
+     ../contrib/zip/src/zip.c
+@@ -1025,7 +1025,6 @@ IF(HUNTER_ENABLED)
+   hunter_add_package(RapidJSON)
+   find_package(RapidJSON CONFIG REQUIRED)
+ ELSE(HUNTER_ENABLED)
+-  INCLUDE_DIRECTORIES( "../contrib/rapidjson/include" )
+   INCLUDE_DIRECTORIES( "../contrib" )
+ ENDIF(HUNTER_ENABLED)
+ 
+@@ -1045,7 +1044,7 @@ if ( MSVC )
+ endif ( MSVC )
+ 
+ IF(NOT HUNTER_ENABLED)
+-  if (UNZIP_FOUND)
++  if (1)
+     SET (unzip_compile_SRCS "")
+   else (UNZIP_FOUND)
+     SET (unzip_compile_SRCS ${unzip_SRCS})
+@@ -1125,7 +1124,7 @@ IF(HUNTER_ENABLED)
+       zip::zip
+   )
+ ELSE(HUNTER_ENABLED)
+-  TARGET_LINK_LIBRARIES(assimp ${ZLIB_LIBRARIES} ${OPENDDL_PARSER_LIBRARIES} ${IRRXML_LIBRARY} )
++  TARGET_LINK_LIBRARIES(assimp ${CONAN_LIBS})
+ ENDIF(HUNTER_ENABLED)
+ 
+ if(ASSIMP_ANDROID_JNIIOSYSTEM)
+--- a/code/Common/BaseImporter.cpp
++++ b/code/Common/BaseImporter.cpp
+@@ -344,7 +344,7 @@ std::string BaseImporter::GetExtension( const std::string& file ) {
+ #ifdef ASSIMP_USE_HUNTER
+ #  include <utf8/utf8.h>
+ #else
+-#  include "../contrib/utf8cpp/source/utf8.h"
++#  include <utf8.h>
```

Comment:
these patches are very invasive, can they be avoided?
(e.g. by creating a proxy utf8.h in contrib directory which will include a real utf.h?)

Reply:
There is no real `utf.h`, could you submit a PR because I don't see how it could be less invasive with this solution.

Reply:
well, if it's the same evil, let's use this approach for now. hope in 5.1 we will remove these patches.